src/Compilers/Core/CommandLine/BuildClient.cs

```diff
@@ -93,13 +93,14 @@ internal int RunCompilation(IEnumerable<string> originalArguments, BuildPaths bu
                         return HandleResponse(buildResponse, parsedArgs, buildPaths);
                     }
                 }
-                catch
+                catch (OperationCanceledException)
                 {
-                    // It's okay, and expected, for the server compilation to fail.  In that case just fall 
-                    // back to normal compilation. 
+                    return CommonCompiler.Succeeded;
```

Comment:
Also, it would be good to add a comment explaining the scenario when this exception could be expected to be thrown by the code in the try and why does it need special handling.